crates/cli-support/src/wit/mod.rs

```diff
@@ -1275,7 +1278,22 @@ impl<'a> Context<'a> {
         };
 
         let mut ret = args.cx.instruction_builder(true);
-        ret.outgoing(&signature.ret)?;
+        if let Instruction::CallExport(_, AuxExportKind::Constructor(ref class)) = call {
+            if let Descriptor::RustStruct(ref name) = signature.ret {
+                if class != name {
```

Comment:
How useful is this check really? I mean all you have to do to bypass it is to wrap your foreign return type inside an `Option` or any other parameterized type and you've gone back to the old behaviour.

Reply:
It should reject `Option<T>` too - only the Rust struct that the constructor's for should be allowed.